pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -462,7 +462,10 @@ func (lbaas *LbaasV2) createOctaviaLoadBalancer(service *corev1.Service, name, c
 		Name:        name,
 		Description: fmt.Sprintf("Kubernetes external service %s/%s from cluster %s", service.Namespace, service.Name, clusterName),
 		Provider:    lbaas.opts.LBProvider,
-		FlavorID:    svcConf.flavorID,
+	}
+
+	if openstackutil.IsOctaviaFeatureSupported(lbaas.lb, openstackutil.OctaviaFeatureFlavors) {
+		createOpts.FlavorID = svcConf.flavorID
 	}
```

Comment:
Could you please move this to `checkService()` function? See `svcConf.allowedCIDR` for an example. The only difference is, by default `FlavorID` should be nil.

Reply:
Moved the check. I am not sure if the `svcConf.flavorID != ""` check I changed this to is really needed since the default is an empty string anyways, but I find it more readable this way